File: crates/deslop/tests/fixtures/fused-golden-rust/shape_alpha.rs
pub struct InventoryDescriptor {
    pub channel: String,
    pub retries: i64,
    pub endpoint: String,
}

impl InventoryDescriptor {
    pub fn compose(&self, tenant: &str) -> String {
        format!("{}/{}/{}?retries=alpha", self.endpoint, tenant, self.channel)
    }
}

File: crates/deslop/tests/fixtures/fused-golden-php/shape_alpha.php
<?php

class InventoryDescriptor
{
    public string $channel = 'alpha-inventory';
    public int $retries = 3;
    public string $endpoint = 'https://alpha.example.com/inventory';

    public function compose(string $tenant): string
    {
        return $this->endpoint . '/' . $tenant . '/' . $this->channel . '?retries=alpha';
    }
}
